.dotnet/src/Custom/OpenAIClient.cs

```diff
@@ -161,4 +167,22 @@ public OpenAIClient(ApiKeyCredential credential = default, OpenAIClientOptions o
     /// </remarks>
     /// <returns> A new <see cref="ModerationClient"/>. </returns>
     public virtual ModerationClient GetModerationClient() => new(_keyCredential, _cachedOptions);
+
+    internal static ClientPipeline CreatePipeline(ApiKeyCredential credential, OpenAIClientOptions options = null)
+    {
+        credential ??= new(Environment.GetEnvironmentVariable("OPENAI_API_KEY") ?? string.Empty);
```

Comment:
so we allow passing null credential to the top level client? i.e. the OpenAIClient.